pkg/plugin/run.go

```diff
@@ -34,11 +34,24 @@ const (
 func (e *AutoscaleEnforcer) startPermitHandler(ctx context.Context, logger *zap.Logger) error {
 	mux := http.NewServeMux()
 	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		logger := logger // copy locally, so that we can add fields and refer to it in defers
+
 		var finalStatus int
 		defer func() {
 			e.metrics.resourceRequests.WithLabelValues(r.RemoteAddr, strconv.Itoa(finalStatus)).Inc()
 		}()
 
+		// Catch any potential panics and report them as 500s
+		defer func() {
+			if err := recover(); err != nil {
+				msg := "request handler panicked"
+				logger.Error(msg, zap.String("error", fmt.Sprint(err)))
+				finalStatus = 500
+				w.WriteHeader(finalStatus)
+				_, _ = w.Write([]byte(msg))
```

Comment:
how about sending not just `msg` here, but  `msg: err`?

Reply:
Hm, yeah. My impression was that it was better from a security perspective to not include the full error message, but tbh maybe it doesn't matter.
wdyt?

Reply:
Hmm, I see your concern. Although this is our internal API, so risk is not high.   

But I guess, it can be left as is, it is easy to grep for logs. Maybe make the message more distinct, like "permit handler panicked" or else.

@areyou1or0 do you have an opinion on whether we should return the full error message, from catch-all exception handler?

Reply:
@Omrigan somehow I didn't get the notification for the mention, just seeing this now. 
Yes, it's best practice to not reveal non-customised error message as this can disclose the details of underlying technology. Low severity but valid.